nncf/experimental/common/pruning/operations.py

```diff
@@ -173,29 +175,16 @@ def _both_dim_blocks_exist(left_idx, right_idx):
                 right = right_dim_groups[input_shape_len - 2]
                 assert len(left) == 1 and len(right) == 1, "multiple groups are not supported"
                 group = PropagationGroup.join_groups(left[0], right[0])
-                cls.add_consumer_block(group, node)
+                pruning_dimension = 1
+                if node.layer_attributes is None:
+                    pruning_dimension = None
+                consumer = ConsumerInfo(node_id=node.node_id, pruning_dimension=pruning_dimension)
+                group.add_consumers({consumer})
+        else:
+            output_mask = node.data.get('output_mask', None)
```

Comment:
IMHO this if is too big, I suggest to move condition bodies (code inside if's) to a separate private method

Reply:
refactored